src/views/open-api-docs-view-v2/utils/get-operation-objects.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// Types
+import type { OpenAPIV3 } from 'openapi-types'
+
+/**
+ * The OpenAPIV3 OperationObject type, with the `path` and `type` (ie GET, POST)
+ * associated with the operation.
+ */
+export type OperationObject = OpenAPIV3.OperationObject & {
+	path: string
+	type: string
+}
+
+/**
+ * Given an OpenAPI document, build an array of operation objects,
+ * each including the properties defined for operations in the OpenAPI spec,
+ * as well as the `path` and `type` (ie GET, POST, etc) of the operation, and
+ * Return the array of operation objects.
+ */
+export function getOperationObjects(
```

Comment:
Gathers operation objects from the provide `openApiDocument`.

We'll probably be able to re-use the result of this function in `getStaticProps` to filter for and then transform the props for a specific operation page.